### DISCUSSION TITLE
Display additional information in diagnostic

### DISCUSSION THREAD

**Discussion by YairHalberstadt** [dotnet/roslyn#46911] [General]
I use a source generator to create an IOC container (see https://github.com/YairHalberstadt/stronginject).

Is there anyway to show a short description by default, but a longer one if the user needs it?

I want to show something like:

> Error whilst resolving dependencies for type a: No available source for type b.

And then a longer description:

> resolution path: a <- c <- ID[] <- D <- b

DiagnosticDescriptor has a field "description" which sounds promising:

```
        //   description:
        //     An optional longer description of the diagnostic.
public DiagnosticDescriptor(string id, string title, string messageFormat, string category, DiagnosticSeverity defaultSeverity, bool isEnabledByDefault, string? description = null, string? helpLinkUri = null, params string[] customTags)
```

However when I add something there I can't see it in Visual Studio, even if all the columns in the error list are shown.

**Comment by jmarolf** [comment]
That is the intended use case for description. @mavasani has anything changed here recently? The error list should still let you expand the diagnostic to see the whole description

**Comment by mavasani** [comment]
@jmarolf Currently we populate the description (and all fields from DiagnosticDescriptor) only for diagnostics reported from intellisense. There is an active item to ensure we do the same for diagnostics reported from build. Likely the case for @YairHalberstadt is for diagnostics coming from build?

**Reply by YairHalberstadt** [reply]
Yea, this is from source generators

**Reply by YairHalberstadt** [reply]
In the build

**Reply by jmarolf** [reply]
> There is an active item to ensure we do the same for diagnostics reported from build.

@mavasani can you link to the issue?

**Reply by YairHalberstadt** [reply]
@mavasani are there any updates on this? Thanks

**Reply by jnm2** [reply]
This is causing a frustrating usability issue in the StrongInject source generator (https://github.com/YairHalberstadt/stronginject/issues/127), and it looks like Yair has no alternative but to wait for the Roslyn bugfix.

**Reply by RikkiGibson** [reply]
Is the implication here that we would be able to see the full description if the source generator were running in IDE and not just during the build?

**Reply by YairHalberstadt** [reply]
It's currently not possible to see the full description of the diagnostic is from build

**Reply by mavasani** [reply]
Error list diagnostics coming from build output only contain the diagnostic ID and the message. Work involved here is map the diagnostic ID for build diagnostic back to compiler or analyzer/generator diagnostic descriptor and then add the title, category and description fields into the diagnostic reported in the error list. Let me open an issue to track this work.

**Reply by mavasani** [reply]
https://github.com/dotnet/roslyn/issues/57377

**Reply by YairHalberstadt** [reply]
Thanks!

As an aside, source generator diagnostics don't seem to assist in the error list without building, which is what causes this whole issue. Is that a bug?

**Reply by mavasani** [reply]
@jasonmalinowski - do we report diagnostics in the IDE from source generators only from build? If so, is there a work item tracking reporting them during live analysis?

**Reply by PathogenDavid** [reply]
> is there a work item tracking reporting them during live analysis?

I'm not Jason, but it's tracked here: https://github.com/dotnet/roslyn/issues/49664

**Reply by jasonmalinowski** [reply]
Yep, that's it. :smile:
